hnn_core/tests/test_gui.py

```diff
@@ -1,8 +1,187 @@
 # Authors: Huzi Cheng <hzcheng15@icloud.com>
-from hnn_core.gui.gui import run_hnn_gui
+import os.path as op
 
+import hnn_core
+import numpy as np
+import pytest
+from hnn_core import Dipole, Network, Params
+from hnn_core.gui.gui import HNNGUI, _init_network_from_widgets
+from hnn_core.network import pick_connection
+from hnn_core.parallel_backends import requires_mpi4py, requires_psutil
 
-def test_run_gui():
-    """Test if main gui function gives proper ipywidget"""
-    gui = run_hnn_gui()
-    assert gui is not None
+
+def test_gui_load_params():
+    """Test if gui loads default parameters properly"""
+    gui = HNNGUI()
+
+    assert isinstance(gui.params, Params)
+
+    print(gui.params)
+    print(gui.params['L2Pyr*'])
+
+
+def test_gui_upload_params():
+    """Test if gui handles uploaded parameters correctly"""
+    gui = HNNGUI()
+    _ = gui.run()
+
+    params_name = 'default.json'
+    hnn_core_root = op.join(op.dirname(hnn_core.__file__))
+    params_fname = op.join(hnn_core_root, 'param', params_name)
+
+    content = b""
+    with open(params_fname, "rb") as f:
+        for line in f:
+            pass
```

Comment:
is this intentional or an artifact of debugging?

Reply:
It's used to load the byte string from the parameter file so that we can simulate the uploading event. But I need to remove the `pass`